src/providers/settings/settings.ts

```diff
@@ -22,11 +66,17 @@ export class SettingsProvider {
     return mergedSettings as GameSettings;
   }
 
+  getStoredSettings = async (): Promise<GameSettings> => {
```

Comment:
I don't like the mixing of async and other async code execution mechanisms, use consistent through codebase